src/app/charts/line-graph.component.ts

```diff
@@ -163,9 +153,11 @@ export class LineGraphComponent implements OnInit, OnDestroy {
 
     // Set axis and line scales
     private setLineScales(): void {
-        // Time scales only recognize annual and daily data
-        var parseTime = D3.timeParse(this.timeFormat);
-        this.extractedData.forEach(d => d.date = parseTime(d.date));
+        // Sort data by date ascending
+        this.extractedData.sort(function(a, b) {return +a.date - +b.date;});
```

Comment:
Me: That seems like a kludge, can we do better?
The Google: https://github.com/Microsoft/TypeScript/issues/2361#issuecomment-82509802
Me: :scream:

Coercion it is!